app/javascript/components/Editor.tsx

```diff
@@ -182,6 +185,7 @@ export default ({
 
       setSubmission(submission.uuid, { ...submission, testRun: testRun })
     },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What's the reasoning for disabling these here? Should they be disabled everywhere?

Reply:
I'm curious about that too. What is the reason we can't be exhaustive here?

Reply:
You are exhaustive here.
The reason why the linter warns here is because it would like to see `[submission, setSubmission]` as deps, as the body of `useCallback` contains those state getter and setter hook-functions. 
But `submission` is wrapped in a `JSON.stringify`, so it no longer recognises it. 

If you don't wrap it in a stringifier, as `SameObject!==SameObject` in JS, it will infinitely redeclare this function, because the hook will think that the `SameObject` has been updated, so it reinvokes itself.

This workaround wouldn't be necessary if React provided an opportunity to add a custom comparison function, but they don't -  despite the fact it is being requested for a long time.

Some of the deps arrays _can_ be more exhaustive, but doesn't really have any effect, because those are static information that would never change on the fly, but are relevant information on the mount. 
e.g. `track.slug` - which in this case doesn't change, because when this page is loaded, track.slug is already determined by Ruby. This could change if everything was handled by React and a big state machine/store.